packages/spectacular-docs/docs/feature-testing/angular-testing-library.md

```diff
@@ -0,0 +1,575 @@
+} from '@tour-of-heroes/crisis-center';
+
+describe('Tour of Heroes: Crisis center integration tests', () => {
+  beforeEach(async () => {
+    result = await render(SpectacularAppComponent, {
+      excludeComponentDeclaration: true,
+      imports: [
+        SpectacularFeatureTestingModule.withFeature({
+          featureModule: CrisisCenterModule,
+          featurePath: crisisCenterPath,
+        }),
+      ],
+    });
+    featureLocation = result.debugElement.injector.get(
+      SpectacularFeatureLocation
+    );
+  });
+
+  let featureLocation: SpectacularFeatureLocation;
+  let result: RenderResult<SpectacularAppComponent, SpectacularAppComponent>;
+
+  it('shows crisis detail when a valid ID is in the URL', async () => {
+    const crisis: Crisis = {
+      id: 1,
+      name: 'Dragon Burning Cities',
+    };
+    await result.navigate(crisisCenterPath);
+
+    fireEvent.click(screen.getByText(crisis.name));
+
+    expect(featureLocation.path()).toBe('~/' + crisis.id);
+  });
+});
+```
+
+## Testing a complete user flow with Spectacular and Angular Testing Library
+
+The following is an example of testing a complete user flow with Spectacular and
+[Angular Testing Library](https://testing-library.com/docs/angular-testing-library/intro/):
+
+```ts {46-64}
+import {
+  SpectacularAppComponent,
+  SpectacularFeatureLocation,
+  SpectacularFeatureRouter,
+  SpectacularFeatureTestingModule,
+} from '@ngworker/spectacular';
+import {
+  fireEvent,
+  render,
+  RenderResult,
+  screen,
+} from '@testing-library/angular';
+import {
+  CrisisCenterModule,
+  crisisCenterPath,
+  CrisisService,
+} from '@tour-of-heroes/crisis-center';
+
+describe('Tour of Heroes: Crisis center integration tests', () => {
+  beforeEach(async () => {
+    result = await render(SpectacularAppComponent, {
+      excludeComponentDeclaration: true,
+      imports: [
+        SpectacularFeatureTestingModule.withFeature({
+          featureModule: CrisisCenterModule,
+          featurePath: crisisCenterPath,
+        }),
+      ],
+    });
+    result.fixture.autoDetectChanges(true);
+    crisisService = result.fixture.debugElement.injector.get(CrisisService);
+    featureLocation = result.fixture.debugElement.injector.get(
+      SpectacularFeatureLocation
+    );
+    featureRouter = result.fixture.debugElement.injector.get(
+      SpectacularFeatureRouter
+    );
+  });
+
+  let crisisService: CrisisService;
+  let featureLocation: SpectacularFeatureLocation;
+  let featureRouter: SpectacularFeatureRouter;
+  let result: RenderResult<SpectacularAppComponent, SpectacularAppComponent>;
+
+  describe('Editing a crisis', () => {
+    it('navigates to the crisis center home with the crisis selected when the change is saved', async () => {
+      const [aCrisis] = crisisService.getCrises().value;
+      await featureRouter.navigate(['~', aCrisis.id]);
+      const newCrisisName = 'Global climate crisis';
+
+      fireEvent.input(screen.getByRole('textbox'), {
```

Comment:
This is correct, the recommended way is to use `userEvent` though - it makes things simpler to write/read and it also sends the correct events to the input, e.g. keydown, keyup, change. 
Docs: https://testing-library.com/docs/ecosystem-user-event/

Reply:
Alright, so this is mainstream and recommended to the point that it would make sense to use and mention here?

Reply:
Yea, it's recommended and popular in the react eco-system.
For ATL, there were some rough edges in earlier versions but it's all good know 👍
I don't have stats, but our projects are using it